File: src/main/java/de/uniReddit/uniReddit/Models/Node.java
package de.uniReddit.uniReddit.Models;


import javax.persistence.*;
import java.util.UUID;

@Entity
@Inheritance(strategy = InheritanceType.JOINED)
@DiscriminatorColumn(name = "OBJECT_TYPE")
@Table(name = "NODES")
public abstract class Node {
    @Id
    @GeneratedValue
    private Long id;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    @Override
    public int hashCode() {
        return getId().hashCode();
    }
}

File: src/main/java/de/uniReddit/uniReddit/Models/Node.kt
package de.uniReddit.uniReddit.Models

import javax.persistence.*

@Entity
@Inheritance(strategy = InheritanceType.JOINED)
@DiscriminatorColumn(name = "OBJECT_TYPE")
@Table(name = "NODES")
abstract class Node{
    @Id @GeneratedValue(strategy = GenerationType.AUTO) var id: Long = 0
}
